app/bundles/PluginBundle/Entity/IntegrationEntity.php

```diff
@@ -0,0 +1,259 @@
+
+        $builder->setTable('integration_entity')
+            ->setCustomRepositoryClass('Mautic\PluginBundle\Entity\IntegrationEntityRepository');
+
+        $builder->addId();
+
+        $builder->addDateAdded();
+
+        $builder->addNullableField('integration', 'array');
+
+        $builder->createField('integrationEntity', 'string')
+            ->columnName('inegration_entity')
+            ->build();
+        $builder->createField('integrationEntity', 'integer')
+            ->columnName('integration_entity_id')
+            ->build();
+        $builder->createField('internalEntity', 'string')
+            ->columnName('internal_entity')
+            ->build();
+        $builder->createField('internalEntityId', 'integer')
+            ->columnName('internal_entity_id')
+            ->build();
+
+        $builder->createField('lastSyncDate', 'datetime')
+            ->columnName('last_sync_date')
+            ->build();
+    }
+
+    /**
+     * Get id
+     *
+     * @return integer
+     */
+    public function getId ()
+    {
+        return $this->id;
+    }
+
+    /**
+     * Set integration
+     *
+     * @param string $integration
+     *
+     * @return string
+     */
+    public function setIntegration ($integration)
+    {
+        $this->integration = $integration;
+
+        return $this;
+    }
+
+    /**
+     * Get name
+     *
+     * @return string
+     */
+    public function getIntegration ()
+    {
+        return $this->integration;
+    }
+
+    /**
+     * Set integrationEntity
+     *
+     * @param string $integrationEntity
+     *
+     * @return integrationEntity
+     */
+    public function setIntegrationEntity ($integrationEntity)
+    {
+        $this->integrationEntity = $integrationEntity;
+    }
+
+    /**
+     * Get integrationEntity
+     *
+     * @return string
+     */
+    public function getIntegrationEntity ()
+    {
+        return $this->integrationEntity;
+    }
+
+    /**
+     * @return int
+     */
+    public function getIntegrationEntityId ()
+    {
+        return $this->integrationEntityId;
+    }
+
+
+    /**
+     * @param mixed $integrationEntityId
+     */
+    public function setIntegrationEntityId ($integrationEntityId)
+    {
+        $this->integrationEntityId = $integrationEntityId;
+    }
+
+    /**
+     * @return string
+     */
+    public function getInternalEntity ()
+    {
+        return $this->integrationEntity;
+    }
+
+    /**
+     * @param mixed string
+     */
+    public function setInternalEntity ($internalEntity)
+    {
+        $this->internalEntity = $internalEntity;
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getInternalEntityId ()
+    {
+        return $this->internalEntityId;
+    }
+
+    /**
+     * @param mixed $isMissing
+     */
+    public function setInternalEntityId ($internalEntityId)
+    {
+        $this->internalEntityId = $internalEntityId;
+    }
+
+    /**
+     * @return mixed
```

Comment:
Is setInternal() and getInternal() used? They aren't defined loadMetadata() or the migration so just validating.